src/main/java/org/mule/extension/db/internal/domain/xa/XADbConnection.java

```diff
@@ -109,4 +113,9 @@ public boolean isStreaming() {
   public void endStreaming() {
     connection.endStreaming();
   }
+
+  @Override
+  public boolean isTransactionActive() {
+    return TransactionCoordination.isTransactionActive();
```

Comment:
why do you need this different mechanism in this case? This will return true even if this connection hasn't joined that transaction

Reply:
Updated to cover all cases, but answering you, I need a different mechanism because the way that the Connection interacts with the TX is different.
When executing inside a XA TX, the only thing that the XADbConnection does is to provide the `XaResource` and all the TX stages of the connection are communicated to the resource and not the connection per se.